### ISSUE TITLE
Dual mining ETH + TON just stopped mining, no errors. HiveOS

### ISSUE THREAD

**Issue by smalla** [Lolliedieb/lolMiner-releases#1122]
relevant portion of the logs:

```
New job received: 0x4d6cd1 Epoch: 471 Difficulty: 4.30G
New job received: 0x0e20eb Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef_NSzfDJI1A3rOM0GQm7xsoUXHTgmdhN5-OrGD8uwL2JHBa received.
New job received: 0x04435f Epoch: 471 Difficulty: 4.30G
New job received: 0x04435f Epoch: 471 Difficulty: 4.30G
---------------------------------------------
Statistics (12:51:57); Uptime: 4h 37m 0s
Algorithm 1: Ethash
Connected to: eth.hiveon.com:4444 (19ms latency)
Algorithm 2: SHA256-TON
Connected to: https://pool.services.tonwhales.com:443 (13ms latency)

      Name     Speed   Pool   Shares    Best    Eff.    Speed     Pool
                Mh/s   Mh/s   A/S/Hw   Share  Mh/s/W     Mh/s     Mh/s
GPU 0 RTX 3070 60.99  66.67  258/0/0  640.8G   0.219  2256.47  2216.22
---------------------------
Total          60.99  66.67  258/0/0  640.8G   0.219  2256.47  2216.22

         Shares   Best    Eff.  Power  CCLK  MCLK  Core  Fan
         A/S/Hw  Share  Mh/s/W      W   MHz   MHz  Temp  Pct
GPU 0   134/0/0  98.1T   8.089  278.5  1995  8055    65   64
---------------------------
Total   134/0/0  98.1T   8.089  278.5
---------------------------------------------
Ethash: Average speed (10s): 60.98 Mh/s
SHA256-TON: Average speed (10s): 2256.26 Mh/s
New job received: 0xd7864f Epoch: 471 Difficulty: 4.30G
New job received: 0x4bd395 Epoch: 471 Difficulty: 4.30G
New job received: 0x4bd395 Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef8JfFUEJhhpRW80_jqD7zzQteH6EBHOzxiOhygRhBdt44YH received.
New job received: 0x6552ea Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef-P_TOdwcCh0AXHhBpICDMxStxHenWdLCDLNH5QcNpwMMn2 received.
New job received: 0x144766 Epoch: 471 Difficulty: 4.30G
New job received: 0x690c0e Epoch: 471 Difficulty: 4.30G
New job received: 0x690c0e Epoch: 471 Difficulty: 4.30G
New job received: 0x690c0e Epoch: 471 Difficulty: 4.30G
Ton pool connection mode: https job polling (ton-miner compatible) (--ton-mode 1)
DNS over HTTPS resolve succeeded
New job received: 0xbd351a Epoch: 471 Difficulty: 4.30G
New job received: 0xbd351a Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef91o4NNTryJ-Cw3sDGt9OTiafmETdVFUMvylQdFPoOxInls received.
New job received: 0xf4ab45 Epoch: 471 Difficulty: 4.30G
New job received: 0xe8d874 Epoch: 471 Difficulty: 4.30G
New job received: 0xe8d874 Epoch: 471 Difficulty: 4.30G
New job received: 0xe8d874 Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef-P_TOdwcCh0AXHhBpICDMxStxHenWdLCDLNH5QcNpwMMn2 received.
New job received: 0x971a6c Epoch: 471 Difficulty: 4.30G
New job received: 0x971a6c Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef8SYc83pm5JkGt0p3TQRkuiM58O9Cr3waUtR9OoFq716uj0 received.
New job received: 0xeb8593 Epoch: 471 Difficulty: 4.30G
New job received: 0x4a83bb Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef8gf1PQy4u2kURl-Gz4LbS29eaN4sVdrVQkPO-JL80VhFww received.
New job received: 0x0af312 Epoch: 471 Difficulty: 4.30G
New job received: 0xe0fcf3 Epoch: 471 Difficulty: 4.30G
New job received: 0xf21a20 Epoch: 471 Difficulty: 4.30G
New job received: 0xf21a20 Epoch: 471 Difficulty: 4.30G
Ethash: Average speed (10s): 5.43 Mh/s
SHA256-TON: Average speed (10s): 0.00 Mh/s
New job received: 0xf21a20 Epoch: 471 Difficulty: 4.30G
New job received: 0xf21a20 Epoch: 471 Difficulty: 4.30G
New job received: 0xc7b8df Epoch: 471 Difficulty: 4.30G
New job received: 0xc7b8df Epoch: 471 Difficulty: 4.30G
New job received: 0xc7b8df Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef_NSzfDJI1A3rOM0GQm7xsoUXHTgmdhN5-OrGD8uwL2JHBa received.
Ton job with difficulty 274.88G and giver Ef-FV4QTxLl-7Ct3E6MqOtMt-RGXMxi27g4I645lw6MTWg0f received.
New job received: 0xcc040a Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef-P_TOdwcCh0AXHhBpICDMxStxHenWdLCDLNH5QcNpwMMn2 received.
New job received: 0x97b7e1 Epoch: 471 Difficulty: 4.30G
New job received: 0x2fea12 Epoch: 471 Difficulty: 4.30G
New job received: 0x2fea12 Epoch: 471 Difficulty: 4.30G
Ton job with difficulty 274.88G and giver Ef8kO6K6Qh6YM4ddjRYYlvVAK7IgyW8Zet-4ZvNrVsmQ4PgP received.
New job received: 0x4f2715 Epoch: 471 Difficulty: 4.30G
New job received: 0x445bf6 Epoch: 471 Difficulty: 4.30G
Ton pool connection mode: https job polling (ton-miner compatible) (--ton-mode 1)
DNS over HTTPS resolve succeeded
Ton job with difficulty 274.88G and giver Ef9iWhwk9GwAXjtwKG-vN7rmXT3hLIT23RBY6KhVaynRrDkx received.
Ton job with difficulty 274.88G and giver Ef8JfFUEJhhpRW80_jqD7zzQteH6EBHOzxiOhygRhBdt44YH received.
New job received: 0xd3d8e7 Epoch: 471 Difficulty: 4.30G
New job received: 0x91caa4 Epoch: 471 Difficulty: 4.30G
Ethash: Average speed (10s): 0.00 Mh/s
SHA256-TON: Average speed (10s): 0.00 Mh/s
Ton job with difficulty 274.88G and giver Ef91o4NNTryJ-Cw3sDGt9OTiafmETdVFUMvylQdFPoOxInls received.
```

Platform: hiveos
lolminer 1.43
card: RTX 3070 non LHR
And its clearly not doing anything because power consumption is dropped as well.
I have restarted the miner after 15 minutes which fixed the problem.

**Comment by tphan26363** [comment]
See like whalepool is rejecting the connection due to waiting for the ETH minning.  Works fine with other pools. Need to work on keeping the connections alive

**Comment by smalla** [comment]
OK than see You till the next release, I will put an auto restart if hashrate drops till than.

**Comment by jgonzis** [comment]
> > OK than see You till the next release, I will put an auto restart if hashrate drops till than.
> 
> Is this feature implemented in 1.51a ? I would like to know if I require programming my own watchdog script to do the job of the miner for the miner.

Watchdog has to be done by each miner or mining OS. We have also put a Watchdog utility for Windows 10.
